src/System.Net.Security/src/System/Net/SecureProtocols/_SslState.cs

```diff
@@ -117,25 +117,17 @@ internal void ValidateCreateContext(bool isServer, string targetHost, SslProtoco
             if (targetHost == null)
             {
                 throw new ArgumentNullException("targetHost");
-            }
-            
-            if (isServer)
-            {
-                enabledSslProtocols &= (SslProtocols)Interop.SChannel.ServerProtocolMask;
-                if (serverCertificate == null)
-                {
-                    throw new ArgumentNullException("serverCertificate");
-                }
-            }
-            else
-            {
-                enabledSslProtocols &= (SslProtocols)Interop.SChannel.ClientProtocolMask;
-            }
```

Comment:
The idea was _sslState, secureChannel etc uber level classes will be independent of interop level functionality. Here code was using Interop.SChannel, which is windows SChannel specific. Instead, now we are directly validating enabledSslProtocols directly & storing the same in secureChannel in place of a masked int. Other logic have been modified as well to support this change.